lib/ruby_smb/server/server_client/session_setup.rb

```diff
@@ -0,0 +1,82 @@
+module RubySMB
+  class Server
+    class ServerClient
+      module SessionSetup
+        #
+        # Setup a new session based on the negotiated dialect. Once session setup is complete, the state will be updated
+        # to :authenticated.
+        #
+        # @param [String] raw_request the session setup request to process
+        def handle_session_setup(raw_request)
+          response = nil
+
+          case metadialect.order
+          when Dialect::ORDER_SMB1
+            request = SMB1::Packet::SessionSetupRequest.read(raw_request)
+            response = do_session_setup_smb1(request)
+          when Dialect::ORDER_SMB2
+            request = SMB2::Packet::SessionSetupRequest.read(raw_request)
+            response = do_session_setup_smb2(request)
+          end
+
+          if response.nil?
+            disconnect!
+          else
+            send_packet(response)
+          end
+
+          nil
+        end
+
+        def do_session_setup_smb1(request)
+          gss_result = process_gss(request.data_block.security_blob)
+          return if gss_result.nil?
+
+          response = SMB1::Packet::SessionSetupResponse.new
+          response.smb_header.pid_low = request.smb_header.pid_low
+          response.smb_header.uid = rand(0x10000)
```

Comment:
Is this `rand` to cut down on computation time by not calling `SecureRandom` all the time?